code/game/objects/items/apc_frame.dm

```diff
@@ -92,7 +92,7 @@
 	if(!..())
 		return
 	var/turf/T = get_turf(on_wall) //the user is not where it needs to be.
-	var/area/A = get_area(T)
+	var/area/A = get_area(user)
```

Comment:
What if you try to make it by telekinesis? I think better check position of frame, instead of user.

Reply:
Took a little while to figure out to admin my DNA mutations but I just gave myself telekinesis for testing and all I can figure out how to do is throw the APC frame around - having it held telekinetically (?) and clicking on the wall just throw it at the wall, and having it in hand and trying to access the wall from distance does nothing.

You're right though, the "no diagonal placement" presumption in the patch is more a case of "the user must be standing where the wire terminal would go", which, for no diagonal or remote placement methods is true.

That said, for case of future compatability, it would be better to find and if necessary refactor out the code that calculates the location of the wire terminal and just uses that.  I'll have a go at this anyway as it's a basically cleaner solution.

Reply:
As i know, when you hit wall by frame it creates dismantled apc in your turf and move image over wall.
Also you can instal frame to wall by hit it with screwdriver when it lay on floor.

Reply:
Tested telekinesis with screwdriver.
If you stand on the APC frame, by the wall, and use the APC frame with a "TK held" screwdriver it will place on the adjacent wall.
If you are standing away from the APC frame and try TK screw the APC frame, it simply creates the "attack" animation on that tile.

So, still can't find any way to remotely deploy an APC frame.

Reply:
Okay.  I had a little play with various ways of doing this, checking out what x and y were, and the various locs of things, when in inventory and in world since both means can be used to place the APC frame.  Went down the rabbit hole of "what is get_area and what is get_step".

It seems that switching "user" to "src" which would pick the location of the APC frame, inventory or in world, would be an alternative resolution, however, when I go look at the existing code earlier on, notably:
`/obj/item/wallframe/proc/try_build(turf/on_wall, mob/user)
        if(get_dist(on_wall,user)>1)
                return
        var/ndir = get_dir(on_wall, user)
        if(!(ndir in GLOB.cardinals))
                return
        var/turf/T = get_turf(user)
        var/area/A = get_area(T)
`
The "super" method (what's DreamMaker speak for this?) which I assume this checks:
`/obj/item/wallframe/apc/try_build(turf/on_wall, user)
        if(!..())
                return
`
is essentially already coded around the "user"'s location, and contains safeguards to ensure the user is exactly 1 tile from the wall location, along with the cardinal location checking...



As a result of this, I suspect leaving my patch to check "user" rather than "src" is at least consistent with the existing internal logic.

Any thoughts?

Reply:
Ok for now.
Maybe i check it later, for telekinesis glory.